# Study 819 — Abnormal-Volume Shock 📊📣

**Do stocks that print abnormally heavy volume go on to drift *up*?**

Garfinkel & Sokobin (2006) argue that trading volume a name's own recent norm cannot
explain is a footprint of **attention / opinion divergence**, and that higher unexplained
volume is followed by a **positive subsequent drift**. We take the self-contained daily
version on a liquid US cross-section (2010-01-04 → 2026-06-30, 50 names):
each name's **standardised abnormal volume** `(V − mean_60)/std_60`, averaged over a 5-day
formation window, then a long-high / short-low sort.

*Numbers below are the frozen headline (`docs/results.md`); the live cells run the fast
synthetic control. Survivorship: current-membership mega-caps — magnitudes are an upper
bound.*


## 1. The idea in one picture

When a name suddenly trades *far more* than its own 60-day norm, that burst is a footprint of fresh information and disagreement about it. If attention and disagreement resolve slowly, the name should keep drifting for a few days. So: measure how many sigmas above its own benchmark each name is trading, buy the loud names, sell the quiet ones.

In [1]:
import numpy as np, pandas as pd
R = dict(spread_bps=0.55, t_nw=0.54, hi_bps=7.44, lo_bps=6.89, gross_sharpe=0.13)
print('long high-avol / short low-avol spread: %+.2f bps/day (NW t = %+.2f)'
      % (R['spread_bps'], R['t_nw']))
print('  high-avol book %+.2f bps vs low-avol book %+.2f bps'
      % (R['hi_bps'], R['lo_bps']))
print('  gross spread Sharpe (before cost): %.2f' % R['gross_sharpe'])

long high-avol / short low-avol spread: +0.55 bps/day (NW t = +0.54)
  high-avol book +7.44 bps vs low-avol book +6.89 bps
  gross spread Sharpe (before cost): 0.13


## 2. Is the sort real? A live synthetic control

We plant the effect in a seeded toy world (`edge>0`: an attention shock that inflates volume *and* lifts the forward return) and check the detector recovers it — and that it stays *silent* on the null (`edge=0`, abnormal volume present but unpriced). No network.

In [2]:
import os, sys
sys.path.insert(0, os.path.abspath('..'))
sys.path.insert(0, os.path.abspath(os.path.join('..','..','..')))
from volume_shock import data, strategy as st
null = st.synthetic_detect(data.synthetic_panel(edge=0.0, seed=819, n_assets=40, n_days=1200))
planted = st.synthetic_detect(data.synthetic_panel(edge=0.0020, seed=819, n_assets=40, n_days=1500))
print('null world   : spread NW t = %+.2f  (should be ~0)' % null['t_nw'])
print('planted world: spread NW t = %+.2f  (should light up)' % planted['t_nw'])

null world   : spread NW t = -0.82  (should be ~0)
planted world: spread NW t = +21.64  (should light up)


## 3. The honest verdict — the famous drift does *not* show up here

On this liquid mega-cap tape the long-high-avol / short-low-avol spread is **+0.55 bps/day** with NW *t* = **+0.54** — the sign matches Garfinkel & Sokobin, but the magnitude is a rounding error: the permutation null centres at 0 (sd 0.86 bps) and the observed value is only ~0.6σ into the right tail (p = 0.27). It even flips sign across the two eras (-0.40 / +0.94). The seeded synthetic control recovers a *planted* attention→drift relation overwhelmingly (*t* = +21.6), so this is a genuine absence on mega-caps, not a dead engine — the disagreement drift is an earnings-window / small-cap effect diluted to nothing here. **Signal: None**, **Tradability: Mirage**.